azurerm/resource_arm_batch_pool.go

```diff
@@ -0,0 +1,518 @@
+	"github.com/hashicorp/terraform/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/response"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmBatchPool() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmBatchPoolCreate,
+		Read:   resourceArmBatchPoolRead,
+		Update: resourceArmBatchPoolUpdate,
+		Delete: resourceArmBatchPoolDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: azure.ValidateAzureRMBatchPoolName,
+			},
+			"resource_group_name": resourceGroupNameSchema(),
+			"account_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateAzureRMBatchAccountName,
+			},
+			"display_name": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"vm_size": {
+				Type:             schema.TypeString,
+				Required:         true,
+				ForceNew:         true,
+				DiffSuppressFunc: suppress.CaseDifference,
+			},
+			"fixed_scale": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"target_dedicated_nodes": {
+							Type:     schema.TypeInt,
+							Optional: true,
+							Default:  1,
+						},
+						"target_low_priority_nodes": {
+							Type:     schema.TypeInt,
+							Optional: true,
+							Default:  0,
+						},
+						"resize_timeout": {
+							Type:     schema.TypeString,
+							Optional: true,
+							Default:  "PT15M",
+						},
+					},
+				},
+			},
+			"auto_scale": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"evaluation_interval": {
+							Type:     schema.TypeString,
+							Optional: true,
+							Default:  "PT15M",
+						},
+						"formula": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+					},
+				},
+			},
+			"storage_image_reference": {
+				Type:     schema.TypeList,
+				Required: true,
+				ForceNew: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+
+						"publisher": {
+							Type:     schema.TypeString,
+							Required: true,
+							ForceNew: true,
+						},
+
+						"offer": {
+							Type:     schema.TypeString,
+							Required: true,
+							ForceNew: true,
+						},
+
+						"sku": {
+							Type:             schema.TypeString,
+							Required:         true,
+							ForceNew:         true,
+							DiffSuppressFunc: suppress.CaseDifference,
+						},
```

Comment:
Could we get at least a `validate.NoEmptyStrings` here?

Reply:
OK.